### PR TITLE
Alcotest_ext: capture and show unchecked test output

### PR DESCRIPTION
**Pull request by mjambon**
This branch adds several important improvements to alcotest_ext. This includes:
* show the output of failed tests at the end of the alcotest_ext output (`run` and `status` subcommands)
* optionally show the output of all tests (`run` and `status` subcommands)

I added a test suite that fails and whose output gets checked by the meta test suite. To preview the effect of failed tests, go to `libs/alcotest_ext`, run `make` and then run `./failing-test`. `make test` runs the meta test suite which is more complicated to interpret.

### CODE REVIEW

File: libs/git_wrapper/Git_wrapper.ml [hunk 1]
```diff
@@ -249,22 +255,21 @@ let run_with_worktree ~commit ?(branch = None) f =
...
-        "--cached";
-        "--name-status";
-        "--no-ext-diff";
-        "-z";
-        "--diff-filter=ACDMRTUXB";
-        "--ignore-submodules";
-        "--relative";
-        commit;
-      ] )
+      cd cwd
```
Comment (aryx) [inline libs/git_wrapper/Git_wrapper.ml:261]: why this change? This seems wrong to me.

File: libs/git_wrapper/Git_wrapper.ml [hunk 2]
```diff
@@ -335,23 +340,27 @@ let status ~cwd ~commit =
     renamed = !renamed;
   }
 
-let is_git_repo cwd =
-  let cmd = (git, [ "-C"; !!cwd; "rev-parse"; "--is-inside-work-tree" ]) in
+let is_git_repo ?cwd () =
+  let cmd = (git, cd cwd @ [ "rev-parse"; "--is-inside-work-tree" ]) in
```
Comment (aryx) [inline libs/git_wrapper/Git_wrapper.ml:344]: This is wrong. The goal o Cmd.t is actually to make things cleaner and forbid shell execution and
be precise on the command executed.
Please revert those changes.

File: src/osemgrep/tests/Test_target_selection.ml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+(*
+   Test target selection on git repos with osemgrep.
+*)
+
+module T = Alcotest_ext
```
Comment (aryx) [inline src/osemgrep/tests/Test_target_selection.ml:5]: This should have been in a separate PR.

File: src/osemgrep/tests/dune [hunk 4]
```diff
@@ -1,7 +1,6 @@
 (library
   (public_name semgrep.osemgrep_tests)
   (name osemgrep_tests)
-  (wrapped false)
```
Comment (aryx) [inline src/osemgrep/tests/dune:4]: please restore (wrapped false).

File: libs/git_wrapper/Git_wrapper.ml (hunk 2 above)
Comment (mjambon) [reply libs/git_wrapper/Git_wrapper.ml:344]: `cd cwd` is not a shell command, it appends the `-C ...` option is `cwd` is specified.
